File: packages/sdk-python/veltrix_sdk/resources/me.py
# -*- coding: utf-8 -*-

from .base import BaseResource


class MeResource(BaseResource):
    """Endpoints scoped to the authenticated user themselves (``/api/me``)."""
    RESOURCE_PATH = "me"

    def get_permissions(self, **kwargs):
        """Retrieves the current user's resolved permission snapshot.

        Corresponds to GET /api/me/permissions.
        """
        return self._http_client.get(f"{self.RESOURCE_PATH}/permissions", **kwargs)

File: packages/sdk-js/src/resources/me.ts
import { BaseResource } from './base';
import { AxiosRequestConfig } from 'axios';

// Endpoints scoped to the authenticated user themselves (/api/me).

interface PermissionSnapshot {
  permissions: Array<{ resource: string; action: string; appId?: string | null }>;
  wildcards: {
    allAll?: boolean;
    resources?: string[];
  };
}

export class MeResource extends BaseResource {
  protected readonly RESOURCE_PATH = 'me';

  /**
   * Returns the current user's resolved permission snapshot.
   * Corresponds to GET /api/me/permissions.
   */
  async getPermissions(config?: AxiosRequestConfig): Promise<PermissionSnapshot> {
    const path = `${this.RESOURCE_PATH}/permissions`;
    return this.httpClient.get<PermissionSnapshot>(path, config);
  }
}
